### PR TITLE
[historical view] Enable country panel for other time aggregates part 1

### PR DESCRIPTION
**Pull request by Kongkille**
_Since #4207 got quite big, I am splitting it out into some different milestone PRs on a temporary main branch_

### Description - Core helpers and constants

This PR adds some core helper functions and constants to be used to format and centralize reusable functionality

### CODE REVIEW

File: web/src/helpers/formatting.js [hunk 1]
```diff
@@ -16,38 +17,89 @@ const formatCo2 = function (d, numDigits) {
...
   // Assume MW input
   if (maxPower < 1)
     return {
-      unit: "kW",
-      formattingFactor: 1e-3
-    }
+      unit: 'kW',
+      formattingFactor: 1e-3,
+    };
   if (maxPower < 1e3)
```
Comment (madsnedergaard) [inline web/src/helpers/formatting.js:32]: Weird that this file wasn't prettified/linted earlier? 🤔 
And also weird that husky/prettier isn't taking effect here and fixing/complaining about lack of brackes in if-else statement...

File: web/src/helpers/formatting.js [hunk 2]
```diff
@@ -2,6 +2,7 @@
 // TODO: remove once refactored
```
Comment (madsnedergaard) [inline web/src/helpers/formatting.js:2]: Ooooh, just answering my own question below about why linting is not working 😄 
Can you remove the `/* eslint-dsiable */` comment and apply linting? :)

File: web/src/helpers/formatting.js [hunk 3]
```diff
@@ -16,38 +17,89 @@ const formatCo2 = function (d, numDigits) {
...
+      console.error(`${time} is not implemented`);
+      return '';
+  }
 };
 
-const formatHourlyDate = function (date, lang) {
-  // formats date object to readable date
-  if (!date) return '';
-  return new Intl.DateTimeFormat(lang, {dateStyle: 'long', timeStyle: 'short' }).format(date);
+const getLocaleUnit = (dateUnit, lang) =>
```
Comment (madsnedergaard) [inline web/src/helpers/formatting.js:58]: Nice hack :)
Were you able to test in older Safari versions?

**Review by madsnedergaard — COMMENTED** [review]
A few small comments, otherwise great! 
I haven't tested it locally yet, will do a through QA when all these PRs are merged if that's okay :)

**Comment by madsnedergaard** [comment]
Also, thanks for splitting this work into multiple PRs 🙏

File: web/src/helpers/formatting.js (hunk 2 above)
Comment (Kongkille) [reply web/src/helpers/formatting.js:2]: Seems a sea monster has been hiding here

image

File: web/src/helpers/formatting.js (hunk 3 above)
Comment (Kongkille) [reply web/src/helpers/formatting.js:67]: No I can't seem to figure out how to do that. Maybe we save that for final QA?

File: web/src/helpers/formatting.js (hunk 2 above)
Comment (madsnedergaard) [reply web/src/helpers/formatting.js]: Haha yeah, ouch 😅

**Pull request merged**